src/wp-includes/rest-api/endpoints/class-wp-rest-font-library-controller.php

```diff
@@ -0,0 +1,453 @@
+					$family_index
+				);
+			}
+
+			if ( isset( $font_family['fontFace'] ) ) {
+				if ( ! is_array( $font_family['fontFace'] ) ) {
+					$error_messages[] = sprintf(
+						// translators: 1: font family index.
+						__( 'Font family [%s] should have fontFace property defined as an array.' ),
+						$family_index
+					);
+					continue;
+				}
+
+				if ( count( $font_family['fontFace'] ) < 1 ) {
+					$error_messages[] = sprintf(
+						// translators: 1: font family index.
+						__( 'Font family [%s] should have at least one font face definition.' ),
+						$family_index
+					);
+				}
+
+				if ( ! empty( $font_family['fontFace'] ) ) {
+					for ( $face_index = 0; $face_index < count( $font_family['fontFace'] ); $face_index++ ) {
+
+						$font_face = $font_family['fontFace'][ $face_index ];
+						if ( ! isset( $font_face['fontWeight'] ) || ! isset( $font_face['fontStyle'] ) ) {
+							$error_messages[] = sprintf(
+								// translators: 1: font family index, 2: font face index.
+								__( 'Font family [%1$s] Font face [%2$s] should have fontWeight and fontStyle properties defined.' ),
+								$family_index,
+								$face_index
+							);
+						}
+
+						if ( isset( $font_face['downloadFromUrl'] ) && isset( $font_face['uploadedFile'] ) ) {
+							$error_messages[] = sprintf(
+								// translators: 1: font family index, 2: font face index.
+								__( 'Font family [%1$s] Font face [%2$s] should have only one of the downloadFromUrl or uploadedFile properties defined and not both.' ),
+								$family_index,
+								$face_index
+							);
+						}
+
+						if ( isset( $font_face['uploadedFile'] ) ) {
+							if ( ! isset( $files[ $font_face['uploadedFile'] ] ) ) {
+								$error_messages[] = sprintf(
+									// translators: 1: font family index, 2: font face index.
+									__( 'Font family [%1$s] Font face [%2$s] file is not defined in the request files.' ),
+									$family_index,
+									$face_index
+								);
+							}
+						}
+					}
+				}
+			}
+		}
+
+		return $error_messages;
+	}
+
+	/**
+	 * Validate input for the install endpoint.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param string          $param The font families to install.
+	 * @param WP_REST_Request $request The request object.
+	 * @return true|WP_Error True if the parameter is valid, WP_Error otherwise.
+	 */
+	public function validate_install_font_families( $param, $request ) {
+		$font_families  = json_decode( $param, true );
+		$files          = $request->get_file_params();
+		$error_messages = $this->get_validation_errors( $font_families, $files );
+
+		if ( empty( $error_messages ) ) {
+			return true;
+		}
+
+		return new WP_Error( 'rest_invalid_param', implode( ', ', $error_messages ), array( 'status' => 400 ) );
+	}
+
+	/**
+	 * Gets the schema for the uninstall endpoint.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return array Schema array.
+	 */
+	public function uninstall_schema() {
+		return array(
+			'fontFamilies' => array(
```

Comment:
As noted elsewhere, params should use snack case.

Reply:
Other schema are cached for performance reasons. Why was this avoided here.

Reply:
Implementing this in this PR: https://github.com/WordPress/gutenberg/pull/54977

Reply:
Is the chaching avoided by writing the schema array inside a function?